pynio/service.py

```diff
@@ -75,6 +95,13 @@ def type(self):
     def status(self):
         return self._status().get('status')
 
+    def delete(self):
+        '''Delete self from instance'''
+        self._instance.services.remove(self._name)
```

Comment:
I think this should only happen if the delete to the rest api is actually successful.